src/backend/distributed/executor/multi_utility.c

```diff
@@ -2733,14 +2807,14 @@ OptionsSpecifyOwnedBy(List *optionList, Oid *ownedByTableId)
  * ErrorIfDistributedRenameStmt errors out if the corresponding rename statement
  * operates on any part of a distributed table other than a column.
  *
- * Note: This function handles only those rename statements which operate on tables.
+ * Note: This function handles RenameStmt applied to relations handed by Citus.
+ * At the moment of writing this comment, it could be either tables or indexes.
  */
 static void
 ErrorIfUnsupportedRenameStmt(RenameStmt *renameStmt)
 {
-	Assert(IsAlterTableRenameStmt(renameStmt));
-
-	if (renameStmt->renameType == OBJECT_TABCONSTRAINT)
+	if (IsAlterTableRenameStmt(renameStmt) &&
+		renameStmt->renameType == OBJECT_TABCONSTRAINT)
```

Comment:
Is there anything stopping us from renaming constraints?

Reply:
All I'm seeing at the moment in the code is this error message in src/backend/distributed/relay/relay_event_utility.c: _"unsafe object type in rename statement"_. I had guessed it's been studied and punt off before already, should we see about adding support for it now?

Reply:
I don't think it's needed in this PR. We might also leave it to another new hire ;).

Reply:
Or I'll do it next to close the gap. I kind of read the comment as “here be dragons”, if it's only “didn't get there yet” that's a different story... for another PR ok.